client/src/app/components/sell-buy-stock/sell-buy-actions.component.spec.ts

```diff
@@ -36,3 +91,53 @@ describe('SellOrBuyActionsComponent', () => {
     expect(component).toBeTruthy();
   });
 });
+
+describe('SellOrBuyActionsComponent unit test', () => {
+  let component: SellOrBuyActionsComponent;
+
+  beforeEach(() => {
+    component = new SellOrBuyActionsComponent(matDialogMock, mockStoreFacade, mockDialogData);
+  });
+
+  it('should return the right stockprice', () => {
+    component.data = null;
+    component.ngOnInit();
+    expect(component.stockPrice).toBe(0);
+    component.data = mockDialogData;
+    expect(component.stockPrice).toBe(mockDialogData.stockInfo.stockValue);
+  });
+
+  it('should return the right stock name', () => {
+    component.data = null;
+    component.ngOnInit();
+    expect(component.stockName).toBe('');
+    component.data = mockDialogData;
+    expect(component.stockName).toBe(component.data.stockInfo.companyName);
+  });
+
+  it('should return the total', () => {
+    component.ngOnInit();
+    expect(component.getTotal()).toBe(0);
+    component.data.stockInfo = mockDialogData.stockInfo;
+    component.quantity = 2;
+    component.isMarketOrder = true;
+    expect(component.getTotal()).toBe(mockDialogData.stockInfo.stockValue * component.quantity);
```

Comment:
It's good that you tested what happens if 'isMarketOrder' is true, but you didn't test what happens if it's false